01-Plot-Raster.Rmd

```diff
@@ -113,67 +126,83 @@ plot(DSM_HARV,
+using a continuous color ramp. This is comparable to a "classified" map in a
+tool like ArcGIS or QGIS.
 
-> R Bloggers Histogram Overview: <a href="http://www.r-bloggers.com/basics-of-histograms/" target="_blank">More 
-on histograms in R</a>
+Unless there is a reason for having specific breaks, we want to use a
+histogram of our data to determine where the breaks in our data should go. We 
+can use `breaks=` to tell `R` to use fewer or more breaks or bins in our 
+histogram. 
+If we name the histogram we can then enter the name an find out much of the
+underlying information in the histogram.  
 
-```{r create-histogram-2-breaks }
+```{r create-histogram-breaks }
 #Plot distribution of raster values 
-hist(DSM_HARV,
-     breaks=2,
+DSMhist<-hist(DSM_HARV,
+     breaks=3,
      main="Histogram Digital Surface Model\nHarvard Forest Field Site",
-     col="wheat3")
+     col="wheat3",  #changes bin color
+     xlab= "Elevation (m)")  #label the x-axis
 
+#Where are breaks and how many pixels in each category?
+DSMhist$breaks
+DSMhist$counts
 ```
 
-Looking at our histogram, R has binned out the data as follows:
+Warning message!?  Remember, the default for the histogram is to include only 
+100,000 values from
+the data set.  We could force it to show all the pixel values or we can use the
+histogram as is and figure that the sample of 100,000 values represents our data
+well. 
+
+Looking at our histogram, `R` has binned out the data as follows:
 
 * 300-350m, 350-400m, 400-450m
 
-We can use those bins to plot our raster data. We will use the `terrain.colors(3)`
-`R` function to create a palette of 3 colors to use in our plot. We can use the 
-`breaks` method to add breaks. Breaks requires a list of numbers formatted as 
-follows:
+We have most of our pixel values in the middle range with a few lower values,
+and very few higher values.  If we wanted a more even number of pixels in each
+bin we could specify different breaks. 
 
-`c(value1,value2,value3)`
+We can use those bins to plot our raster data. We will use the 
+`terrain.colors(3)` function to create a palette of 3 colors to use in our plot.
 
-You can include as many or as few breaks as you'd like but you need atleast 3 
-breaks to create a plot of any value.
+We can use the `breaks` method to add breaks.  To specify where the breaks
+occur, we need to use the following syntax: `breaks=c(value1,value2,value3)`.
+We can include as few or many breaks as we'd like.
 
 ```{r plot-with-breaks }
-
-#Create a color palette of 3 colors
-myCol = terrain.colors(3)
-
 #plot using breaks.
 plot(DSM_HARV, 
-     breaks = c(300, 350, 400, 450), 
-     col = myCol,
+     breaks = c(350, 300, 400, 450), 
+     col = terrain.colors(3),
      main="NEON Digital Surface Model\nHarvard Forest Field Site")
 
 ```
 
-We can label the X and Y axis of our plot too. 
+###Formatting Color & Plot Axes
+If we're going to be creating multiple plots and we want to easily be able to
```

Comment:
Nice. Please edit

If we need to create multiple plots using the same color palette, we can create an `R` object...